smartsim/settings/batchSettings.py

```diff
@@ -44,12 +44,73 @@
 
 
 class BatchSettings(BaseSettings):
+    """The BatchSettings class manages the configuration and execution of batch jobs
+    across the resources of an HPC system.
+
+    BatchSettings is designed to be extended by a BatchArguments child class that
+    corresponds to the scheduler provided during initialization. The supported schedulers
+    are Slurm, PBS, and LSF. Using the BatchSettings class, users can:
+
+    - Set the scheduler type of a batch job.
+    - Configure batch arguments and environment variables.
+    - Access and modify custom batch arguments.
+    - Update environment variables.
+    - Retrieve information associated with the ``BatchSettings`` object.
+        - The scheduler value (BatchSettings.scheduler).
+        - The derived BatchArguments child class (BatchSettings.scheduler_args).
+        - The set environment variables (BatchSettings.env_vars).
+        - A formatted output of set batch arguments (BatchSettings.format_batch_args).
+    """
+
     def __init__(
         self,
         batch_scheduler: t.Union[SchedulerType, str],
```

Comment:
What do you think about just using the name ``scheduler`` for this parameter?

Reply:
better!